e2e/testsuite/testsuite.go

```diff
@@ -0,0 +1,286 @@
+package testsuite
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"strings"
+	"testing"
+	"time"
+
+	dockerclient "github.com/docker/docker/client"
+	"github.com/strangelove-ventures/ibctest"
+	"github.com/strangelove-ventures/ibctest/chain/cosmos"
+	"github.com/strangelove-ventures/ibctest/ibc"
+	"github.com/strangelove-ventures/ibctest/relayer"
+	"github.com/strangelove-ventures/ibctest/testreporter"
+	"github.com/stretchr/testify/suite"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zaptest"
+
+	"github.com/cosmos/ibc-go/v4/e2e/testconfig"
+)
+
+const (
+	SourceRelayerName      = "srcRly"
+	DestinationRelayerName = "dstRly"
+)
+
+// E2ETestSuite has methods and functionality which can be shared among all test suites.
+type E2ETestSuite struct {
+	suite.Suite
+	chainPairs     map[string]chainPair
+	logger         *zap.Logger
+	Client         *dockerclient.Client
+	network        string
+	startRelayerFn func(relayer ibc.Relayer)
+}
+
+// chainPair is a pairing of two chains which will be used in a test.
+type chainPair struct {
+	srcChain, dstChain *cosmos.CosmosChain
+}
+
+// ChainOptions stores chain configurations for the chains that will be
+// created for the tests. They can be modified by passing ChainOptionConfiguration
+// to E2ETestSuite.GetChains.
+type ChainOptions struct {
+	SrcChainConfig *ibc.ChainConfig
+	DstChainConfig *ibc.ChainConfig
+}
+
+// ChainOptionConfiguration enables arbitrary configuration of ChainOptions.
+type ChainOptionConfiguration func(options *ChainOptions)
+
+// GetChains returns a src and dst chain that can be used in a test. The pair returned
+// is unique to the current test being run. Note: this function does not create containers.
+func (s *E2ETestSuite) GetChains(chainOpts ...ChainOptionConfiguration) (*cosmos.CosmosChain, *cosmos.CosmosChain) {
+	if s.chainPairs == nil {
+		s.chainPairs = map[string]chainPair{}
+	}
+	cp, ok := s.chainPairs[s.T().Name()]
```

Comment:
I have a mild preference for avoiding one/two letter variable naming unless it is contained within a for loop. I think I'd prefer to just use `chainPair`

Reply:
I tend to use short variable names when the scope is limited (loops as you mentioned or small functions). I don't feel very strongly about it, so I'm happy to rename to `chainPair` in this case.

Reply:
Agree with both, I usually prefer descriptive/informative names. For example `rly` over `r` says much more to me. 
But in general it depends on the scope and lifespan of the variable, for small loops and functions I feel fine about one or two letter namings. In my opinion its usually fine if the type/variable is inferred from the function name that was invoked too. The exception is func receivers, I usually prefer short-hand for them.

Reply:
oh whoops, I meant to tag you on this [comment](https://github.com/cosmos/ibc-go/pull/1650#discussion_r919963422) @damiannolan. Agree with your response though